lib/ansible/modules/crypto/openssl_privatekey.py

```diff
@@ -190,8 +192,8 @@ def generate(self, module):
                             self.mode = AnsibleModule._symbolic_mode_to_octal(st, self.mode)
                         except ValueError as e:
                             module.fail_json(msg="%s" % to_native(e), exception=traceback.format_exc())
-                os.chmod(self.path, self.mode)
                 privatekey_file = os.open(self.path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
+                os.chmod(self.path, self.mode)
```

Comment:
You're not setting the owner of the file until the end of this module (through `module.set_fs_attributes_if_different()`). I'd rather only change the mode to `self.mode` after writing the sensitive data, so that any early abort without cleanup will result in a key only being readable by the user `root` (and not any groups, for example).

Reply:
@thomwiggers Thank you for your review! I'll trying to fix the code based on your advice :)

Reply:
@thomwiggers ping

Reply:
> @thomwiggers with root, do you mean the current user used by Ansible?

Yes

Reply:
I don't see why `chmod` should be necessary considering `module.set_fs_attributes_if_different` is used.

Reply:
I agree. The only reason to call it at that moment is to make sure that the file isn't readable by everyone (depending on umask). In any case, the `chmod` should be called before `write`, otherwise this probably isn't correct.